### PR TITLE
Support for some more hdf5 types

### PR DESCRIPTION
**Pull request by hodger**
Adds support for all variations of: 

- `MAP_STRING_PAIR_DOUBLE_MAP_INT_DOUBLE`
- `MAP_INT_MAP_STRING_DOUBLE`
- `PAIR_INT_PAIR_STRING_STRING`
- `VECTOR_PAIR_INT_PAIR_STRING_STRING`
- `MAP_STRING_VECTOR_PAIR_INT_PAIR_STRING_STRING`

Includes a small `gentypesystem.py` fix (among bugfixes for the hdf5 backend generator)

@scopatz

### CODE REVIEW

File: cyclus/gentypesystem.py [hunk 1]
```diff
@@ -576,6 +576,7 @@ def convert_to_cpp(self, x, t):
         'pyfirst = {firstexpr}\n'
         '{secondname} = {var}.second\n'
         '{secondbody}\n'
+        'pyfirst = {firstexpr}\n'
```
Comment (scopatz) [inline cyclus/gentypesystem.py]: Whoops! Mea culpa!

File: src/query_backend.h [hunk 2]
```diff
@@ -307,6 +307,20 @@ enum DbTypes {
...
+  VECTOR_PAIR_INT_PAIR_VL_STRING_VL_STRING, // ["std::vector<std::pair<int, std::pair<std::string, std::string>>>", 1, ["HDF5"], ["VECTOR", ["PAIR", "INT", ["PAIR", "VL_STRING", "VL_STRING"]]], false]
+  VL_VECTOR_PAIR_INT_PAIR_STRING_STRING, // ["std::vector<std::pair<int, std::pair<std::string, std::string>>>", 1, ["HDF5"], ["VL_VECTOR", ["PAIR", "INT", ["PAIR", "STRING", "STRING"]]], true]
+  VL_VECTOR_PAIR_INT_PAIR_VL_STRING_STRING, // ["std::vector<std::pair<int, std::pair<std::string, std::string>>>", 1, ["HDF5"], ["VL_VECTOR", ["PAIR", "INT", ["PAIR", "VL_STRING", "STRING"]]], true]
+  VL_VECTOR_PAIR_INT_PAIR_STRING_VL_STRING, // ["std::vector<std::pair<int, std::pair<std::string, std::string>>>", 1, ["HDF5"], ["VL_VECTOR", ["PAIR", "INT", ["PAIR", "STRING", "VL_STRING"]]], true]
+  VL_VECTOR_PAIR_INT_PAIR_VL_STRING_VL_STRING, // ["std::vector<std::pair<int, std::pair<std::string, std::string>>>", 1, ["HDF5"], ["VL_VECTOR", ["PAIR", "INT", ["PAIR", "VL_STRING", "VL_STRING"]]], true]
```
Comment (scopatz) [inline src/query_backend.h]: You have to add these to the end.  This enum is append only.

**Comment by scopatz** [comment]
A couple of minor issues, but otherwise looks great! Thanks @hodger!

File: src/query_backend.h (hunk 2 above)
Comment (hodger) [inline src/query_backend.h]: Ah, okay. So after the resource types?

Comment (scopatz) [inline src/query_backend.h]: Correctamundo, Ace.

**Comment by hodger** [comment]
@scopatz I moved those types to the end- any other issues you noticed?

**Comment by scopatz** [comment]
LGTM! Thanks!

**Pull request merged**